test/ZkMinterModTriggerV1.t.sol

```diff
@@ -1,343 +1,350 @@
-    vm.prank(user);
+    vm.prank(minter);
     vm.expectRevert(
-      abi.encodeWithSignature("ZkCappedMinterV2__CapExceeded(address,uint256)", address(badTrigger), 1000 ether)
+      abi.encodeWithSelector(
+        ZkMinterModTriggerV1.ZkMinterModTriggerV1__InvalidRecipient.selector, _to, address(minterTrigger)
+      )
     );
-    badTrigger.mint(1000 ether); // Try to mint 1000 ether (exceeds cap of 500 ether)
+    minterTrigger.mint(_to, _amount);
+  }
+
+  function testFuzz_RevertIf_CalledByNonMinter(address _nonMinter, address _to, uint256 _amount) public {
+    vm.assume(_nonMinter != minter && _nonMinter != admin);
+
+    vm.prank(_nonMinter);
+    vm.expectRevert(_formatAccessControlError(_nonMinter, MINTER_ROLE));
+    minterTrigger.mint(_to, _amount);
   }
 
-  function test_CallWithCustomCallData() public {
-    // Expect the event with the fixed amount from callData
-    vm.expectEmit(true, false, false, true, address(target));
-    emit TransferProcessed(address(trigger), 500 ether);
+  function testFuzz_RevertIf_MintAfterContractIsPaused(address _caller, address _to, uint256 _amount) public {
+    vm.prank(admin);
+    minterTrigger.pause();
 
-    // Call initiateCall
-    vm.prank(user);
-    trigger.mint(500 ether);
+    vm.prank(_caller);
+    vm.expectRevert("Pausable: paused");
+    minterTrigger.mint(_to, _amount);
+  }
 
-    // Verify token transfer
-    assertEq(token.balanceOf(address(trigger)), 0);
-    assertEq(token.balanceOf(address(target)), 500 ether);
+  function testFuzz_RevertIf_MintAfterContractIsClosed(address _caller, address _to, uint256 _amount) public {
+    vm.prank(admin);
+    minterTrigger.close();
 
-    // Verify allowance was set and consumed
-    assertEq(token.allowance(address(trigger), address(target)), 0);
+    vm.prank(_caller);
+    vm.expectRevert(ZkMinterV1.ZkMinter__ContractClosed.selector);
+    minterTrigger.mint(_to, _amount);
   }
 }
 
-contract MintFromZkCappedMinter is ZkMinterModTriggerV1Test {
-  function setUp() public virtual override {
-    // Call parent setUp first
+contract Trigger is ZkMinterModTriggerV1Test {
+  function setUp() public override {
     super.setUp();
+    vm.prank(admin);
+    minterTrigger.grantRole(MINTER_ROLE, caller);
+    vm.deal(address(caller), 100 ether);
   }
 
-  function test_MintFromCappedMinterAndInitiateCall() public {
-    // Verify initial state
-    assertEq(token.balanceOf(address(trigger)), 0);
-    assertEq(token.balanceOf(address(target)), 0);
+  function test_TriggersSuccessfully() public {
+    assertEq(mockTarget.value(), 0);
+    assertEq(mockTarget.called(), false);
+    assertEq(address(minterTrigger).balance, 0);
 
-    // Expect the TransferProcessed event with the fixed amount
-    vm.expectEmit(true, false, false, true, address(target));
-    emit TransferProcessed(address(trigger), 500 ether);
+    vm.prank(caller);
+    minterTrigger.trigger{value: 100 ether}();
```

Comment:
Can this value be fuzzed?

Reply:
Yes can be fuzzed, and no it doesn't revert if msg.value is above the value set in constructor. https://github.com/zksync-association/zkminters/pull/33/commits/a76c4480f88b32dbce2a5a521c491195ceef0817